mass_editing/migrations/14.0.1.0.0/pre-migrate.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (C) 2020 - Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if openupgrade.table_exists(env.cr, "mass_editing"):
+        migrate_mass_editing(env)
+
+
+def migrate_mass_editing(env):
+    """ Migrates mass.editing to ir.actions.server """
+    # Remove FK (mass_editing_id)
+    openupgrade.remove_tables_fks(env.cr, ["mass_editing_line"])
+    # Create ir.actions.server for each mass.editing
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE ir_act_server
+        ADD COLUMN __legacy_mass_editing_id__       int4,
+        ADD COLUMN __legacy_ref_ir_act_window_id__  int4
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO ir_act_server (
+            __legacy_mass_editing_id__,
+            __legacy_ref_ir_act_window_id__,
+            name,
+            type,
+            usage,
+            binding_type,
+            activity_user_type,
+            state,
+            model_id,
+            model_name,
+            domain
+        )
+        SELECT
+            me.id,
+            me.ref_ir_act_window_id,
+            COALESCE(me.action_name, me.name),
+            'ir.actions.server',
+            'ir_actions_server',
+            'action',
+            'specific',
+            'mass_edit',
+            me.model_id,
+            mo.model,
+            me.domain
+        FROM mass_editing me
+        LEFT JOIN ir_model mo ON (me.model_id = mo.id)
+        """,
+    )
+    # Delete fields that no longer exist (ondelete didn't exist before)
+    openupgrade.logged_query(
+        env.cr,
+        """
+        DELETE FROM mass_editing_line
+        WHERE field_id IS NULL
+        OR field_id NOT IN (SELECT id FROM ir_model_fields)
+        """,
+    )
+    # Migrate mass.editing.line
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE mass_editing_line
+        ADD COLUMN server_action_id int4
```

Comment:
Do this on post-migration for avoiding to create the column on advance, or use `add_field` openupgrade method

Reply:
done